apps/builder/app/shared/style-object-model.test.ts

```diff
@@ -0,0 +1,213 @@
+};
+
+test("use cascaded style when specified and fallback to initial value", () => {
+  const styles: StyleDecl[] = [
+    {
+      breakpointId: "base",
+      styleSourceId: "bodyLocal",
+      property: "width",
+      value: { type: "unit", value: 10, unit: "px" },
+    },
+  ];
+  const model: StyleObjectModel = {
+    styleSourcesByInstanceId: new Map([["body", ["bodyLocal"]]]),
+    styleByStyleSourceId: getStyleByStyleSourceId(styles),
+  };
+  const styleSelector: StyleSelector = {
+    instanceSelector: ["body"],
+  };
+  // cascaded property
+  expect(
+    getComputedStyleDecl({ model, styleSelector, property: "width" })
+      .computedValue
+  ).toEqual({ type: "unit", unit: "px", value: 10 });
+  // initial for not inherited property
+  expect(
+    getComputedStyleDecl({ model, styleSelector, property: "height" })
+      .computedValue
+  ).toEqual({ type: "keyword", value: "auto" });
+  // initial for inherited property
+  expect(
+    getComputedStyleDecl({ model, styleSelector, property: "color" })
+      .computedValue
+  ).toEqual({ type: "keyword", value: "black" });
+});
+
+test("support initial keyword", () => {
+  const styles: StyleDecl[] = [
+    {
+      breakpointId: "base",
+      styleSourceId: "bodyLocal",
+      property: "width",
+      value: { type: "keyword", value: "initial" },
+    },
+  ];
+  const model: StyleObjectModel = {
+    styleSourcesByInstanceId: new Map([["body", ["bodyLocal"]]]),
+    styleByStyleSourceId: getStyleByStyleSourceId(styles),
+  };
+  const styleSelector: StyleSelector = {
+    instanceSelector: ["body"],
+  };
+  expect(
+    getComputedStyleDecl({ model, styleSelector, property: "width" })
+      .computedValue
+  ).toEqual({ type: "keyword", value: "auto" });
+});
+
+test("support inherit keyword", () => {
+  const styles: StyleDecl[] = [
+    {
+      breakpointId: "base",
+      styleSourceId: "level2Local",
+      property: "width",
+      value: { type: "unit", value: 10, unit: "px" },
+    },
+    {
+      breakpointId: "base",
+      styleSourceId: "level3Local",
+      property: "width",
+      value: { type: "keyword", value: "inherit" },
+    },
+    {
+      breakpointId: "base",
+      styleSourceId: "level1Local",
+      property: "height",
+      value: { type: "unit", value: 20, unit: "px" },
+    },
+    {
+      breakpointId: "base",
+      styleSourceId: "level3Local",
+      property: "height",
+      value: { type: "keyword", value: "inherit" },
+    },
+  ];
+  const model: StyleObjectModel = {
+    styleSourcesByInstanceId: new Map([
+      ["level3", ["level3Local"]],
+      ["level2", ["level2Local"]],
+      ["level1", ["level1Local"]],
+    ]),
+    styleByStyleSourceId: getStyleByStyleSourceId(styles),
+  };
+  const styleSelector: StyleSelector = {
+    instanceSelector: ["level3", "level2", "level1"],
+  };
+  // should inherit declared value
+  expect(
+    getComputedStyleDecl({ model, styleSelector, property: "width" })
```

Comment:
nit:  getStyleDecl or computeStyleDecl would be shorter

Reply:
Mirroring browser api